rosidl_generator_type_description/rosidl_generator_type_description/__init__.py

```diff
@@ -397,35 +407,31 @@ def field_type_type_name(ftype: definition.AbstractType) -> str:
     return value_type_name + name_suffix
 
 
-def field_type_type_id(ftype: definition.AbstractType) -> Tuple[str, int]:
+def field_type_type_id(ftype: definition.AbstractType) -> int:
     return FIELD_TYPE_NAME_TO_ID[field_type_type_name(ftype)]
 
 
 def field_type_capacity(ftype: definition.AbstractType) -> int:
     if isinstance(ftype, definition.AbstractNestedType):
-        if ftype.has_maximum_size():
-            try:
-                return ftype.maximum_size
-            except AttributeError:
-                return ftype.size
+        if isinstance(ftype, definition.Array):
+            return ftype.size
+        elif isinstance(ftype, definition.BoundedSequence):
+            return ftype.maximum_size
     return 0
 
 
-def field_type_string_capacity(ftype: definition.AbstractType) -> int:
+def field_type_string_capacity(ftype: definition.AbstractType) -> Union[int, str]:
     value_type = ftype
     if isinstance(ftype, definition.AbstractNestedType):
         value_type = ftype.value_type
 
     if isinstance(value_type, definition.AbstractGenericString):
-        if value_type.has_maximum_size():
-            try:
-                return value_type.maximum_size
-            except AttributeError:
-                return value_type.size
+        if isinstance(value_type, (definition.BoundedString, definition.BoundedWString)):
+            return value_type.maximum_size
     return 0
```

Comment:
Again I think this would be simpler as:

```
if isinstance(value_type, (definition.BoundedString, definition.BoundedWString)):
    return value_type.maximum_size

return 0
```